Description: Add clGetKernelSubGroupInfoKHR to _cl_icd_dispatch table

As ocl-icd-libopencl1's clGetExtensionFunctionAddress(forPlatform)
tries the dispatch table before the ICD's clGetExtensionFunctionAddress,
attempting to call functions that are in ocl-icd-libopencl1's table
but not the ICD's table will usually crash.

Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Forwarded: https://lists.freedesktop.org/archives/beignet/2016-October/008055.html

--- a/src/cl_khr_icd.c
+++ b/src/cl_khr_icd.c
@@ -17,7 +17,7 @@
 #include <ocl_icd.h>
 
 #include "cl_platform_id.h"
-
+#include "CL/cl_intel.h" // for clGetKernelSubGroupInfoKHR
 /* The interop functions are not implemented in Beignet */
 #define CL_GL_INTEROP(x) NULL
 /* OpenCL 1.2 is not implemented in Beignet */
@@ -168,7 +168,24 @@ struct _cl_icd_dispatch const cl_khr_icd
   (void *) NULL,
   (void *) NULL,
   (void *) NULL,
-  (void *) NULL
+  (void *) NULL,
+#if (OCL_ICD_IDENTIFIED_FUNCTIONS > 110)
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  (void *) NULL,
+  clGetKernelSubGroupInfoKHR,
+#endif
 #endif
 };
 
